### PR TITLE
Feature(Domain Implementation)

### LINKED ISSUE #1: I think im retarded
Now i m sure

### PR DESCRIPTION
**Pull request by Dachshund77**
So I made the majority o the Domain implmentation. Was a bit of slave work but here we go. I didnt quite finish the asp backend communication with xamarin tonight, so this was left out for now.

### CODE REVIEW

File: CodeBase/Commons/Domain/User.cs [hunk 1]
```diff
@@ -49,44 +49,172 @@ public User(User other)
...
+        private List<string> InvalidUserNameReasons()
         {
-            throw new NotImplementedException();
+            //Init values
+            List<string> returnList = new List<string>();
+
+            //Test
+            if (PlainPassword == null)
+            {
+                returnList.Add("UserName may not be negative.");
```
Comment (InLe0) [inline CodeBase/Commons/Domain/User.cs:102]: Odd Message

File: CodeBase/Commons/Domain/User.cs [hunk 2]
```diff
@@ -49,44 +49,172 @@ public User(User other)
...
+        private List<string> InvalidEmailReasons()
         {
-            throw new NotImplementedException();
+            //Init values
+            List<string> returnList = new List<string>();
+
+            //Test
+            if (Email == null)
+            {
+                returnList.Add("Email may not be negative.");
```
Comment (InLe0) [inline CodeBase/Commons/Domain/User.cs]: Odd formulation, but ok-ish enaugh

**Review by InLe0 — APPROVED** [review]

**Comment by Dachshund77** [comment]
@InLe0 I fixed the typos, thanks for checking my code :D

**Review by maximilienviallon — APPROVED** [review]
looks good to me

**Pull request merged**